src/System.Net.Http/src/System/Net/Http/Managed/ConnectionCloseDuplexStream.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal sealed partial class HttpConnection : IDisposable
+    {
+        private sealed class ConnectionCloseDuplexStream : HttpContentDuplexStream
```

Comment:
Can we call this something like "ConnectionUpgradeStream" -- that seems more descriptive.

Reply:
Actually, since I think we want to use this for CONNECT as well, can we call it something like RawConnectionStream?

Reply:
Ok